LIBVmotor0.11/LIBVmotor0.11.ino

```diff
@@ -1,172 +1,249 @@
+  Wire.onReceive(receiveEvent);       // I2C receive data event
+
+  
+  EEPROM_read(); //Get Setting from EEPROM
+  BPM_Timing();  //Calc breath timing
+
+  
+  motor.init();  //Motor Setup
+
+  //Initialize the Arms
+  while(true){
+    motor.setPWM(-200);
+    if(digitalRead(HALL1_PIN)) break;;
+  }
+  motor.setPWM(0);
+  delay(500);
+
+  //Reset Encoder & Timer
+  Enc.write(0);
+  xEn = Enc.read();
+  timer = millis();
+  xt=millis();
+}
+
+
+void loop() {
+  //readSerial();  //Testing cmd from Serial
+
+  if(!breathing){
+    goPos = ZERO; //go to HOME position
+    if(BREATH_BUT.push()) breathing=true; //Run breathing
+  }else{
+    if(BREATH_BUT.holdTime() >= brth_hold_time) brth_hold_stat=true;
+    if(brth_hold_stat && BREATH_BUT.push()){
+      brth_hold_stat=false;
+      breathing=false;
+    }
+
+    switch(venStat){
+      case pushIN:
+        goPos = TV2RPB;
+        speedMov = speedIn;
+        if(millis()-timer>=timeIn || abs(goPos-aPos)<=0.04){
+          timer = millis();
+          venStat = pushOUT;
+        }
+        break;
+      case pushOUT:
+        goPos = ZERO;
+        speedMov = speedOut;
+        if(millis()-timer>=timeOut){
+          timer = millis();
+          venStat = pushRest;
+        }
+        break;
+      case pushRest:
+        if(millis()-timer>=timeRest){
+          timer = millis();
+          venStat = pushIN;
+        }
+        break;
+    }
+
+  }
+
+  
+
+  MotorControl();
+}
+
+void MotorControl(){
+  t = millis();
+  dt = t-xt;
+  
+  /*-- Check speed and position every 10ms --*/
+  if(dt >= 10){
+    xt=t;                               //Store time to last timer
+    En = Enc.read();                    //Get encoder value
+    dPos =(En-xEn)/(mGear*mPpr*mPulse); //Calculate round for 10ms
+    aPos += dPos;                       //Actual Position
+    aSpeed = 1000.0*dPos/dt;            //Actual speed in Round per second (RPS)
+    xEn=En;                             //Passing last encoder value
+  }
+  
+  if(abs(goPos-aPos)<1.1 && abs(goPos-aPos)>0.05) goSpeed = speedMov;
+  else goSpeed = Ramp.cmd(goPos, aPos, dt); 
+  //goSpeed = Ramp.cmd(goPos, aPos, dt);
+  goSpeed *= 1.3;         //Mantain the 30% speed, PiD drop to correct the error (bad tuning) or do the better PiD tuning
+  cmd = PiD.cmd(goSpeed, aSpeed, dt);    //PiD generate digital control to motor
+  if(brth_hold_stat) cmd = 0;
+  motor.setPWM(cmd);
+}
+
+void readSerial() {
+  if(Serial.available()) {
+    String inString = Serial.readString();
+    goPos = inString.toFloat();
+    Serial.println("dspeed: "+(String)goPos);
+  }
+}
+
+//Alarm sound and LED, silent button to silence the alarm for period of time
+void Alarm(bool _stat, int type){
+  //TODO: define alarm type
+  //TODO: trigger the alarm with _stat
+  //TODO: silent button
+}
+
+//Calculate the Breath timing from Setting
+void BPM_Timing(){
+  timeIn = TI;
+  timeOut = (TI*IE*0.5)/2.0;
```

Comment:
Are you dividing this by 2 because it only takes half the time for the arms to retract, and the other half they are resting? That's worth commenting :) 
We can talk about that in person also.